packages/gce-disk-expand/src/expandfs-lib.sh

```diff
@@ -60,32 +60,31 @@ blkid_get_fstype() (
     echo $ID_FS_TYPE
 )
 
+sgdisk_get_label() {
+    local root="$1"
+    [ -z "$root" ] && return 0
+
+    if sgdisk -p "$root" | grep -q "Found invalid GPT and valid MBR"; then
+        echo "mbr"
+    else
+        echo "gpt"
+    fi
+}
 
-# Checks for and corrects the end-of-disk GPT backup block in case of expanded
-# disk.
-parted_fix_gpt() {
+sgdisk_fix_gpt() {
   local disk="$1"
   [ -z "$disk" ] && return
 
-  if parted -sm "$disk" print 2>&1 | grep -q "fix the GPT"; then
-    # Running parted prompts the user to fix this condition, but only does so in
-    # the interactive exception handler. In order to pass input we must use the
-    # hidden triple-dash flag and pass both print and Fix arguments. `print`
-    # alone will not perform the fix, but `Fix` alone will fail the argument
-    # parser.
-    parted -m ---pretend-input-tty "$disk" print Fix >/dev/null 2>&1 </dev/null
-    parted -m ---pretend-input-tty "$disk" print Fix >/dev/null 2>&1 </dev/null
-    if parted -sm "$disk" print 2>&1 | grep -q "fix the GPT"; then
-      echo "Failed to fix the GPT."
-      return 1
-    fi
-  fi
+  local label=$(sgdisk_get_label "$disk")
+  [ "$label" != "gpt" ] && return
+
+  sgdisk --move-second-header "$disk"
```

Comment:
I believe you need to re-probe the partitions after this. At least that was what I found (partprobe).

Reply:
what did you find? we aren't changing partitions here so i didn't find this to be the case

Reply:
It was several weeks back but I believe that you would still get the error that the header was wrong unless the kernel was forced to re-read the partition table. I believe this was when I was trying to do a check with sgdisk however.